src/plugins/discourse/fetch.js

```diff
@@ -283,6 +294,55 @@ export class Fetcher implements Discourse {
     const json = await response.json();
     return json.user_actions.map(parseLike);
   }
+
+  async topicsBumpedSince(sinceMs: number): Promise<Topic[]> {
+    const topics: Topic[] = [];
+    let lastUnpinnedTimestamp: number = Infinity;
+    let morePages: boolean = true;
+    let page: number = 0;
+
+    // Keep going till we've found timestamps older than sinceMs.
+    while (lastUnpinnedTimestamp >= sinceMs && morePages) {
+      const response = await this._fetch(
+        `/latest.json?order=activity&ascending=false&page=${page}`
+      );
+      failIfMissing(response);
+      failForNotOk(response);
+      const {users, topic_list: topicList} = await await response.json();
+
+      // Having the same amount of results as expected by pagination, assume there's another page.
+      morePages = topicList.per_page == topicList.topics.length;
+
+      // The API doesn't provide usernames for each topic author.
+      // Instead there's a top-level mapping to IDs, which we store here.
+      const usernamesById: Map<number, string> = new Map(
+        users.map((u) => [u.id, u.username])
+      );
+
+      for (const jsonTopic of topicList.topics) {
+        const topic = parseLatestTopic(usernamesById, jsonTopic);
+
+        // Due to how pinning works, we may have some topics in here that weren't bumped past `sinceMs`.
+        // Filter those out now.
+        if (topic.bumpedMs > sinceMs) {
+          topics.push(topic);
+        }
+
+        // Make sure we ignore pinned topics for this value, as pinned topics move to the top,
+        // and are unhelpful in knowing whether we should fetch another page.
+        if (!jsonTopic.pinned) {
+          lastUnpinnedTimestamp = Math.min(
```

Comment:
It is because it is an assumption about the API's sorting. Already an example of within-page breaking of this assumption exists in the form of pinning. This changes the semantic from, whichever timestamp the last topic in the array that isn't pinned has, to the lowest timestamp of unpinned topics in this array.

So long as the API doesn't change out from under us, they would be the same and we can do without the Math.min.